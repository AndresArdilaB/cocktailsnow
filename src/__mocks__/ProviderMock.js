import React from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { Router } from 'react-router-dom';
import { createBrowserHistory } from 'history';
import { testState } from '../frontend/initialState';
import reducer from '../frontend/reducers';

const store = createStore(reducer, testState);
const history = createBrowserHistory();

const ProviderMock = ({ children }) => (
  <Provider store={store}>
    <Router history={history}>
      {children}
    </Router>
  </Provider>
);

export default ProviderMock;
