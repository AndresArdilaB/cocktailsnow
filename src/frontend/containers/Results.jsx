import React, { useState, useEffect } from 'react';
import { connect } from 'react-redux';
import Filters from '../components/Filters';
import { getData } from '../utils/getApiData';
import CocktailList from '../components/CocktailList';
// import '../assets/styles/App.scss';

const Results = ({ filters, cocktailName, order, match: { params } }) => {
  const [cocktails, setCocktails] = useState([]);

  useEffect(() => {
    const textFilter = cocktailName ? cocktailName.toUpperCase() : null;

    const filter = cocktails.filter((f) => f.strDrink.toUpperCase().match(textFilter));

    if (filter.length > 0 && cocktailName !== '') {
      setCocktails(filter);
    } else {
      getData(params.param, params.value)
        .then((data) => {
          const newOrder = data;

          if (order === 'asc') {
            newOrder.sort();
          } else if (order === 'desc') {
            newOrder.reverse();
          }

          setCocktails(newOrder);
        });
    }

  }, [params.param, params.value, cocktailName, order]);

  return (
    <>
      <Filters filters={filters} />
      {
        !cocktails ?
          <div className='alert'>No cocktails with this filter.</div> :
          <CocktailList items={cocktails} />
      }
    </>
  );
};

const mapStateToProps = ({ filters, cocktailName, order }) => {
  return {
    filters,
    cocktailName,
    order,
  };
};

export default connect(mapStateToProps, null)(Results);
