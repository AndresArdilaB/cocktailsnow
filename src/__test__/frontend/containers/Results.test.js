import React from 'react';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import ProviderMock from '../../../__mocks__/ProviderMock';
import Results from '../../../frontend/containers/Results';

describe('<Results />', () => {

  test('Component render', () => {
    const results = shallow(
      <ProviderMock>
        <Results />
      </ProviderMock>,
    );

    expect(results.length).toEqual(1);
  });

});

describe('Results Snapshot', () => {

  test('Testing the Results UI component', () => {
    const results = create(
      <ProviderMock>
        <Results
          match={{ params: {} }}
        />
      </ProviderMock>,
    );
    expect(results.toJSON()).toMatchSnapshot();
  });
});
