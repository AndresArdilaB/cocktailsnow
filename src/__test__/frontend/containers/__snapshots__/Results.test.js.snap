// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results Snapshot Testing the Results UI component 1`] = `
Array [
  <section
    className="filters__main"
  >
    <div
      className="filters__title"
    >
      Find your cocktail by some of this parameters
    </div>
    <section
      className="filters"
    >
      <div
        className="container"
      >
        <select
          className="select"
          onChange={[Function]}
        >
          <option
            value=""
          >
            Categories
          </option>
          <option
            id="test"
            value="test"
          >
            Test
          </option>
        </select>
      </div>
      <div
        className="container"
      >
        <select
          className="select"
          onChange={[Function]}
        >
          <option
            value=""
          >
            Glasses
          </option>
          <option
            id="test"
            value="test"
          >
            Test
          </option>
        </select>
      </div>
      <div
        className="container"
      >
        <select
          className="select"
          onChange={[Function]}
        >
          <option
            value=""
          >
            Ingredients
          </option>
          <option
            id="test"
            value="test"
          >
            Test
          </option>
        </select>
      </div>
      <div
        className="container"
      >
        <select
          className="select"
          onChange={[Function]}
        >
          <option
            value=""
          >
            Type
          </option>
          <option
            id="test"
            value="test"
          >
            Test
          </option>
        </select>
      </div>
    </section>
    <input
      className="search"
      onChange={[Function]}
      placeholder="Find by name in this list..."
      type="text"
    />
    <div
      className="order_container"
    >
      Order by name:
      <span
        className="order_button"
        onClick={[Function]}
        role="button"
        tabIndex={0}
      >
        Asc
      </span>
      |
      <span
        className="order_button"
        onClick={[Function]}
        role="button"
        tabIndex={0}
      >
        Desc
      </span>
    </div>
  </section>,
  <section
    className="container_list"
  >
    <div
      className="cocktails_list"
    />
  </section>,
]
`;
